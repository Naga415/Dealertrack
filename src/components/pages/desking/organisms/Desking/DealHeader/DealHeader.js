import React from 'react';
import PropTypes from 'prop-types';
import Button from '@coxautokc/fusion-ui-components/lib/Button';

// eslint-disable-next-line react/prefer-stateless-function
class DealHeader extends React.PureComponent {
  constructor(){
    super();
    this.state = {
      showCreditScoreModal: false,
      showZipcodeModal: false
    };
    this.getLink = this.getLink.bind(this);
  }

  getLink = (value, id, handler) => {
    return (
      <Button bsStyle="link" htmlId={id} className="deal-header__button" onClick={handler}>{value}</Button>
    );
  }

  openCloseCreditScoreDialog = event => {
    event.preventDefault();
    this.setState(prevState => ({showCreditScoreModal: !prevState.showCreditScoreModal}));
  }

  openCloseZipcodeModel = event => {
    event.preventDefault();
    this.setState(prevState => ({showZipcodeModal: !prevState.showZipcodeModal}));
  }

  render() {
    const zipCodeNode = this.getLink(this.props.zipcode, 'buttonZipcode', this.openCloseZipcodeModel);
    const creditscoreNode = this.getLink(this.props.creditscore, 'buttonCreditscore', this.openCloseCreditScoreDialog);
    return (
      <div id={this.props.htmlId}>
        Deal is based on Zipcode{zipCodeNode}and a Credit Score of{creditscoreNode}
      </div>
    );
  }
}

DealHeader.defaultProps = {
  creditscore : 850
};


DealHeader.propTypes = {
  zipcode: PropTypes.string.isRequired,
  creditscore: PropTypes.number,
  htmlId: PropTypes.string.isRequired
};

export default DealHeader;
