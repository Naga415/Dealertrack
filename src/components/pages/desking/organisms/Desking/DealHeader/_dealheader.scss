$dealHeader :'.deal-header';

#{$dealHeader} {
  &__button {
    padding: 0 4px !important;
  }

  &__header-col {
    padding: 0;
  }

  &__actionbar-col {
    padding: 0;

    .action-bar {
      margin-left: 70px;
    }
  }

  &__menubar-col {
    padding: 0;
  }
}
