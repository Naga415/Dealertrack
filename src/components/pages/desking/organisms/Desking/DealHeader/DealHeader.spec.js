import React from 'react';
import renderer from 'react-test-renderer';
import DealHeader from './DealHeader';
import { shallow, mount } from 'enzyme';
import Button from '@coxautokc/fusion-ui-components/lib/Button';

function setupshallow (zipcode, creditscore) {
  const props = {
    htmlId:"DealerHeader",
    zipcode,
    creditscore: creditscore || undefined
  };
  return shallow(<DealHeader {...props} />);
}

function setupmount (zipcode, creditscore) {
  const props = {
    htmlId:"DealerHeader",
    zipcode,
    creditscore: creditscore || undefined
  };
  return mount(<DealHeader {...props} />);
}

describe('DealHeader', ()=>{
  it('renders <DealHeader />', () =>{
    const component = renderer.create(<DealHeader htmlId="DealHeader" zipcode="66203" creditscore={650} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render 2 <Button /> components', () => {
    const component = setupshallow('66203',650);
    expect(component.find(Button).length).toBe(2);
  });

  it('should render 850 as a default value for credit score', () => {
    const component = setupmount('66203');
    expect(component.props().creditscore).toEqual(850);
  });

  it('should render 66202 for zipcode and 750 for creditscore', () => {
    const props = {
      zipcode : '66203',
      creditscore: 750
    };
    const component = setupmount(props.zipcode, props.creditscore);
    expect(component.props().creditscore).toEqual(props.creditscore);
    expect(component.props().zipcode).toEqual(props.zipcode);
  });

  it('should show/hide the respective dialog when the Button is clicked', ()=> {
    const props = {
      htmlId:'DealerHeader',
      zipcode: '66201',
      creditscore: 750
    };
    const component = mount(<DealHeader {...props} />);
    component.find('#buttonZipcode').simulate('click');
    expect(component.state().showZipcodeModal).toEqual(true);
    component.find('#buttonZipcode').simulate('click');
    expect(component.state().showZipcodeModal).toEqual(false);
    component.find('#buttonCreditscore').simulate('click');
    expect(component.state().showCreditScoreModal).toEqual(true);
    component.find('#buttonCreditscore').simulate('click');
    expect(component.state().showCreditScoreModal).toEqual(false);
  });
});
