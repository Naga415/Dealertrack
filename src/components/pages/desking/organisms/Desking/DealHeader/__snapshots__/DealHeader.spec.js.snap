// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DealHeader renders <DealHeader /> 1`] = `
<div
  id="DealHeader"
>
  Deal is based on Zipcode
  <button
    className="deal-header__button btn btn-link"
    disabled={false}
    href={undefined}
    id="buttonZipcode"
    onClick={[Function]}
    type="button"
  >
    66203
  </button>
  and a Credit Score of
  <button
    className="deal-header__button btn btn-link"
    disabled={false}
    href={undefined}
    id="buttonCreditscore"
    onClick={[Function]}
    type="button"
  >
    650
  </button>
</div>
`;
